config/crd/patches/validation_in_linodeclusters.yaml

```diff
@@ -0,0 +1,9 @@
+# The following patch adds additional constraints after the built-in name validation for the CRD
```

Comment:
could we use kubebuilder [validations](https://book.kubebuilder.io/reference/markers/crd-validation) for these usecases, or are they complex enough that they need to be built right into the CRD?

Reply:
Unfortunately not 😢 See 0d41fd3ba77c03a5895ce30d1e9612c8588fcbe7 😢

Reply:
oh, is that because we are validating the names of the resources and not some other spec field?

Reply:
Pretty much. I was able to "hack" around this limitation, but it breaks the validation messaging from the API server: https://kubernetes.slack.com/archives/CAR30FCJZ/p1711475793972459